src/control/lib/hardware/sysfs/provider.go

```diff
@@ -0,0 +1,48 @@
+//
+// (C) Copyright 2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package sysfs
+
+import (
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+)
+
+// NewProvider creates a new SysfsProvider.
+func NewProvider() *Provider {
+	return &Provider{
+		root: "/sys",
+	}
+}
+
+// SysfsProvider provides system information from sysfs.
+type Provider struct {
+	root string
+}
+
+// GetNetDevClass fetches the network device class for the given network interface.
+func (s *Provider) GetNetDevClass(dev string) (hardware.NetDevClass, error) {
+	if dev == "" {
+		return 0, errors.New("device name required")
+	}
+
+	if s.root == "" {
+		s.root = "/sys"
+	}
```

Comment:
minor: Might be better to have a `getRoot()` function so that this check doesn't need to be duplicated as more methods are added to implement other interfaces.

Reply:
Good idea. Fixed.